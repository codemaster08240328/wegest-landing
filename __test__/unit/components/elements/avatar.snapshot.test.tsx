import renderer from 'react-test-renderer'
import Avatar from '@/components/elements/avatar'

describe('<Avatar> snapshot', () => {
  it('renders without props', () => {
    expect(renderer.create(<Avatar />).toJSON()).toMatchSnapshot()
  })

  it('renders with image', () => {
    expect(
      renderer.create(<Avatar src='/images/avatar-test.jpg' />).toJSON(),
    ).toMatchSnapshot()
  })

  it('renders with image and name', () => {
    expect(
      renderer
        .create(<Avatar src='/images/avatar-test.jpg' name='John Smith' />)
        .toJSON(),
    ).toMatchSnapshot()
  })

  it('renders with name', () => {
    expect(
      renderer.create(<Avatar name='John Smith' />).toJSON(),
    ).toMatchSnapshot()
  })

  it('renders with a class', () => {
    expect(
      renderer.create(<Avatar className='class-test' />).toJSON(),
    ).toMatchSnapshot()
  })
})
