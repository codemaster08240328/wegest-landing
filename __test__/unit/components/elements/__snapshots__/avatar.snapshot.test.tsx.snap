// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar> snapshot renders with a class 1`] = `
<div
  className="MuiAvatar-root Component-root-1 Component-root-11 MuiAvatar-circle class-test MuiAvatar-colorDefault Component-colorDefault-2 Component-colorDefault-12"
  data-testid="el-avatar"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiAvatar-fallback"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
    />
  </svg>
</div>
`;

exports[`<Avatar> snapshot renders with image 1`] = `
<div
  className="MuiAvatar-root Component-root-1 Component-root-5 MuiAvatar-circle"
  data-testid="el-avatar"
>
  <img
    className="MuiAvatar-img"
    src="/images/avatar-test.jpg"
  />
</div>
`;

exports[`<Avatar> snapshot renders with image and name 1`] = `
<div
  className="MuiAvatar-root Component-root-1 Component-root-7 MuiAvatar-circle"
  data-testid="el-avatar"
>
  <img
    alt="John Smith"
    className="MuiAvatar-img"
    src="/images/avatar-test.jpg"
  />
</div>
`;

exports[`<Avatar> snapshot renders with name 1`] = `
<div
  className="MuiAvatar-root Component-root-1 Component-root-9 MuiAvatar-circle MuiAvatar-colorDefault Component-colorDefault-2 Component-colorDefault-10"
  data-testid="el-avatar"
>
  JS
</div>
`;

exports[`<Avatar> snapshot renders without props 1`] = `
<div
  className="MuiAvatar-root Component-root-1 Component-root-3 MuiAvatar-circle MuiAvatar-colorDefault Component-colorDefault-2 Component-colorDefault-4"
  data-testid="el-avatar"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiAvatar-fallback"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
    />
  </svg>
</div>
`;
